<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="shinytest2-tm_file_viewer" timestamp="2024-05-10T13:35:02Z" hostname="c4f7783ae627" tests="13" skipped="0" failures="0" errors="0" time="17.995">
    <testcase time="4.262" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Initializes_without_errors_and_shows_files_tree_specified_in_input_path_argument"/>
    <testcase time="0.0200000000000005" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Initializes_without_errors_and_shows_files_tree_specified_in_input_path_argument"/>
    <testcase time="0.0140000000000002" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Initializes_without_errors_and_shows_files_tree_specified_in_input_path_argument"/>
    <testcase time="0.04" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Initializes_without_errors_and_shows_files_tree_specified_in_input_path_argument"/>
    <testcase time="0.00499999999999989" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Initializes_without_errors_and_shows_files_tree_specified_in_input_path_argument"/>
    <testcase time="4.173" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_image_file"/>
    <testcase time="0.00600000000000023" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_image_file"/>
    <testcase time="4.22" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_text_file"/>
    <testcase time="0.00699999999999967" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_text_file"/>
    <testcase time="0.00200000000000067" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_text_file"/>
    <testcase time="0.145999999999999" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_text_file"/>
    <testcase time="4.957" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_url"/>
    <testcase time="0.143000000000001" classname="shinytest2_tm_file_viewer" name="e2e_tm_file_viewer_Shows_selected_url"/>
  </testsuite>
  <testsuite name="shinytest2-tm_front_page" timestamp="2024-05-10T13:35:20Z" hostname="c4f7783ae627" tests="3" skipped="0" failures="0" errors="3" time="305.918">
    <testcase time="101.981" classname="shinytest2_tm_front_page" name="e2e_tm_front_page_Initializes_without_errors_and_check_html_elements">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_front_page.R:26:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:35:21.11 Start AppDriver initialization
{shinytest2} R  info   13:35:21.17 Starting Shiny app
{shinytest2} R  info   13:35:22.39 Creating new ChromoteSession
{shinytest2} R  info   13:35:22.43 Navigating to Shiny app
{shinytest2} R  info   13:35:22.50 Injecting shiny-tracer.js
{chromote}   JS info   13:35:22.51 shinytest2; jQuery not found
{chromote}   JS info   13:35:22.51 shinytest2; Loaded
{shinytest2} R  info   13:35:22.51 Waiting for Shiny to become ready
{chromote}   JS info   13:35:22.61 shinytest2; jQuery found
{chromote}   JS info   13:35:22.61 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:35:22.67 shinytest2; Connected
{chromote}   JS info   13:35:22.72 shinytest2; shiny:busy
{shinytest2} R  info   13:35:22.73 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:35:22.74 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:37:02.76 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:4329
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_front_page() at test-shinytest2-tm_front_page.R:26:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_front_page.R:7:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
    <testcase time="101.949" classname="shinytest2_tm_front_page" name="e2e_tm_front_page_Verify_the_module_displays_tables">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_front_page.R:50:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:37:03.08 Start AppDriver initialization
{shinytest2} R  info   13:37:03.13 Starting Shiny app
{shinytest2} R  info   13:37:04.36 Creating new ChromoteSession
{shinytest2} R  info   13:37:04.40 Navigating to Shiny app
{shinytest2} R  info   13:37:04.47 Injecting shiny-tracer.js
{chromote}   JS info   13:37:04.48 shinytest2; jQuery not found
{chromote}   JS info   13:37:04.48 shinytest2; Loaded
{shinytest2} R  info   13:37:04.48 Waiting for Shiny to become ready
{chromote}   JS info   13:37:04.57 shinytest2; jQuery found
{chromote}   JS info   13:37:04.57 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:37:04.64 shinytest2; Connected
{chromote}   JS info   13:37:04.68 shinytest2; shiny:busy
{shinytest2} R  info   13:37:04.71 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:37:04.71 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:38:44.72 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:3546
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_front_page() at test-shinytest2-tm_front_page.R:50:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_front_page.R:7:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
    <testcase time="101.988" classname="shinytest2_tm_front_page" name="e2e_tm_front_page_Verify_the_module_displays_metadata">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_front_page.R:62:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:38:45.03 Start AppDriver initialization
{shinytest2} R  info   13:38:45.09 Starting Shiny app
{shinytest2} R  info   13:38:46.31 Creating new ChromoteSession
{shinytest2} R  info   13:38:46.35 Navigating to Shiny app
{shinytest2} R  info   13:38:46.46 Injecting shiny-tracer.js
{chromote}   JS info   13:38:46.47 shinytest2; jQuery not found
{chromote}   JS info   13:38:46.47 shinytest2; Loaded
{shinytest2} R  info   13:38:46.47 Waiting for Shiny to become ready
{chromote}   JS info   13:38:46.56 shinytest2; jQuery found
{chromote}   JS info   13:38:46.56 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:38:46.62 shinytest2; Connected
{chromote}   JS info   13:38:46.66 shinytest2; shiny:busy
{shinytest2} R  info   13:38:46.69 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:38:46.69 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:40:26.71 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:4279
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_front_page() at test-shinytest2-tm_front_page.R:62:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_front_page.R:7:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
  </testsuite>
  <testsuite name="shinytest2-tm_misssing_data" timestamp="2024-05-10T13:40:26Z" hostname="c4f7783ae627" tests="10" skipped="0" failures="0" errors="5" time="509.601">
    <testcase time="0.118000000000052" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Initializes_without_errors"/>
    <testcase time="101.803" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Initializes_without_errors">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_misssing_data.R:41:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:40:27.03 Start AppDriver initialization
{shinytest2} R  info   13:40:27.04 Starting Shiny app
{shinytest2} R  info   13:40:28.27 Creating new ChromoteSession
{shinytest2} R  info   13:40:28.30 Navigating to Shiny app
{shinytest2} R  info   13:40:28.37 Injecting shiny-tracer.js
{chromote}   JS info   13:40:28.38 shinytest2; jQuery not found
{chromote}   JS info   13:40:28.38 shinytest2; Loaded
{shinytest2} R  info   13:40:28.39 Waiting for Shiny to become ready
{chromote}   JS info   13:40:28.47 shinytest2; jQuery found
{chromote}   JS info   13:40:28.47 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:40:28.53 shinytest2; Connected
{chromote}   JS info   13:40:28.57 shinytest2; shiny:busy
{shinytest2} R  info   13:40:28.61 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:40:28.61 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:42:08.63 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:4499
{shiny}      R  stderr ----------- Initializing reporter_previewer_module
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_missing_data() at test-shinytest2-tm_misssing_data.R:41:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_misssing_data.R:17:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
    <testcase time="0.105999999999995" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Default_settings_and_visibility_of_the_summary_graph"/>
    <testcase time="101.789" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Default_settings_and_visibility_of_the_summary_graph">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_misssing_data.R:62:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:42:08.94 Start AppDriver initialization
{shinytest2} R  info   13:42:08.96 Starting Shiny app
{shinytest2} R  info   13:42:10.18 Creating new ChromoteSession
{shinytest2} R  info   13:42:10.21 Navigating to Shiny app
{shinytest2} R  info   13:42:10.29 Injecting shiny-tracer.js
{chromote}   JS info   13:42:10.30 shinytest2; jQuery not found
{chromote}   JS info   13:42:10.30 shinytest2; Loaded
{shinytest2} R  info   13:42:10.30 Waiting for Shiny to become ready
{chromote}   JS info   13:42:10.48 shinytest2; jQuery found
{chromote}   JS info   13:42:10.48 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:42:10.48 shinytest2; Connected
{chromote}   JS info   13:42:10.50 shinytest2; shiny:busy
{shinytest2} R  info   13:42:10.51 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:42:10.51 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:43:50.52 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:3545
{shiny}      R  stderr ----------- Initializing reporter_previewer_module
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_missing_data() at test-shinytest2-tm_misssing_data.R:62:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_misssing_data.R:17:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
    <testcase time="0.100000000000023" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Check_default_settings_and_visibility_of_the_combinations_graph_and_encodings"/>
    <testcase time="101.832" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Check_default_settings_and_visibility_of_the_combinations_graph_and_encodings">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_misssing_data.R:91:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:43:50.82 Start AppDriver initialization
{shinytest2} R  info   13:43:50.84 Starting Shiny app
{shinytest2} R  info   13:43:52.07 Creating new ChromoteSession
{shinytest2} R  info   13:43:52.10 Navigating to Shiny app
{shinytest2} R  info   13:43:52.21 Injecting shiny-tracer.js
{chromote}   JS info   13:43:52.22 shinytest2; jQuery not found
{chromote}   JS info   13:43:52.22 shinytest2; Loaded
{shinytest2} R  info   13:43:52.22 Waiting for Shiny to become ready
{chromote}   JS info   13:43:52.31 shinytest2; jQuery found
{chromote}   JS info   13:43:52.31 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:43:52.38 shinytest2; Connected
{chromote}   JS info   13:43:52.42 shinytest2; shiny:busy
{shinytest2} R  info   13:43:52.43 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:43:52.44 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:45:32.45 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:4159
{shiny}      R  stderr ----------- Initializing reporter_previewer_module
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_missing_data() at test-shinytest2-tm_misssing_data.R:91:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_misssing_data.R:17:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
    <testcase time="0.0950000000000273" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Validate_functionality_and_UI_response_for_By_Variable_Levels_"/>
    <testcase time="101.811" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Validate_functionality_and_UI_response_for_By_Variable_Levels_">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_misssing_data.R:122:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:45:32.75 Start AppDriver initialization
{shinytest2} R  info   13:45:32.77 Starting Shiny app
{shinytest2} R  info   13:45:34.00 Creating new ChromoteSession
{shinytest2} R  info   13:45:34.03 Navigating to Shiny app
{shinytest2} R  info   13:45:34.12 Injecting shiny-tracer.js
{chromote}   JS info   13:45:34.13 shinytest2; jQuery not found
{chromote}   JS info   13:45:34.13 shinytest2; Loaded
{shinytest2} R  info   13:45:34.13 Waiting for Shiny to become ready
{chromote}   JS info   13:45:34.24 shinytest2; jQuery found
{chromote}   JS info   13:45:34.24 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:45:34.31 shinytest2; Connected
{shinytest2} R  info   13:45:34.34 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:45:34.34 shinytest2; Waiting for Shiny to be stable
{chromote}   JS info   13:45:34.36 shinytest2; shiny:busy
{shinytest2} R  info   13:47:14.36 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:7186
{shiny}      R  stderr ----------- Initializing reporter_previewer_module
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_missing_data() at test-shinytest2-tm_misssing_data.R:122:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_misssing_data.R:17:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
    <testcase time="0.100000000000023" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Validate_By_Variable_Levels_table_values"/>
    <testcase time="101.847" classname="shinytest2_tm_misssing_data" name="e2e_tm_missing_data_Validate_By_Variable_Levels_table_values">
      <error type="error" message="Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout,  ('test-shinytest2-tm_misssing_data.R:152:3')">Error in `app_initialize(self, private, app_dir = app_dir, ..., load_timeout = load_timeout, 
    timeout = timeout, wait = wait, expect_values_screenshot_args = expect_values_screenshot_args, 
    screenshot_args = screenshot_args, check_names = check_names, 
    name = name, variant = variant, view = view, height = height, 
    width = width, seed = seed, clean_logs = clean_logs, shiny_args = shiny_args, 
    render_args = render_args, options = options)`: Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable


i You can inspect the failed AppDriver object via `rlang::last_error()$app`
i AppDriver logs:
{shinytest2} R  info   13:47:14.67 Start AppDriver initialization
{shinytest2} R  info   13:47:14.69 Starting Shiny app
{shinytest2} R  info   13:47:15.91 Creating new ChromoteSession
{shinytest2} R  info   13:47:15.94 Navigating to Shiny app
{shinytest2} R  info   13:47:16.06 Injecting shiny-tracer.js
{chromote}   JS info   13:47:16.07 shinytest2; jQuery not found
{chromote}   JS info   13:47:16.07 shinytest2; Loaded
{shinytest2} R  info   13:47:16.07 Waiting for Shiny to become ready
{chromote}   JS info   13:47:16.16 shinytest2; jQuery found
{chromote}   JS info   13:47:16.16 shinytest2; Waiting for shiny session to connect
{chromote}   JS info   13:47:16.22 shinytest2; Connected
{chromote}   JS info   13:47:16.26 shinytest2; shiny:busy
{shinytest2} R  info   13:47:16.29 Waiting for Shiny to become idle for 200ms within 1e+05ms
{chromote}   JS info   13:47:16.29 shinytest2; Waiting for Shiny to be stable
{shinytest2} R  info   13:48:56.31 Error while initializing AppDriver:
                                   Shiny app did not become stable in 1e+05ms.
                                   Message: An error occurred while waiting for Shiny to be stable
                                   Caused by error in `app_wait_for_idle()`:
                                   ! An error occurred while waiting for Shiny to be stable
{shiny}      R  stderr ----------- Loading required package: shiny
{shiny}      R  stderr ----------- Attempting to load the environment 'package:nestcolor'
{shiny}      R  stderr ----------- Loading required package: ggmosaic
{shiny}      R  stderr ----------- Loading required package: ggplot2
{shiny}      R  stderr ----------- Loading required package: teal
{shiny}      R  stderr ----------- Loading required package: teal.data
{shiny}      R  stderr ----------- Loading required package: teal.code
{shiny}      R  stderr ----------- Loading required package: teal.slice
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- You are using teal version 0.15.2.9036
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Attaching package: 'teal'
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- The following objects are masked from 'package:teal.slice':
{shiny}      R  stderr ----------- 
{shiny}      R  stderr -----------     as.teal_slices, teal_slices
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Loading required package: teal.transform
{shiny}      R  stderr ----------- Running application in test mode.
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- createTcpServer: address already in use
{shiny}      R  stderr ----------- 
{shiny}      R  stderr ----------- Listening on http://127.0.0.1:3081
{shiny}      R  stderr ----------- Initializing reporter_previewer_module
{shiny}      R  stderr ----------- Warning: Error in if: the condition has length &gt; 1
{shiny}      R  stderr -----------   14: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   12: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   11: &lt;Anonymous&gt;
{shiny}      R  stderr -----------   10: base::do.call
{shiny}      R  stderr -----------    9: base::saveRDS
{shiny}      R  stderr -----------    8: base::withCallingHandlers
{shiny}      R  stderr -----------    7: doTryCatch
{shiny}      R  stderr -----------    6: tryCatchOne
{shiny}      R  stderr -----------    5: tryCatchList
{shiny}      R  stderr -----------    2: tryCatchList
{shiny}      R  stderr -----------    1: base::tryCatch


Caused by error in `app_initialize_()`:
! Shiny app did not become stable in 1e+05ms.
Message: An error occurred while waiting for Shiny to be stable
Caused by error in `app_wait_for_idle()`:
! An error occurred while waiting for Shiny to be stable
Backtrace:
     x
  1. +-teal.modules.general (local) app_driver_tm_missing_data() at test-shinytest2-tm_misssing_data.R:152:3
  2. | \-teal.modules.general:::init_teal_app_driver(...) at test-shinytest2-tm_misssing_data.R:17:3
  3. |   +-testthat::with_mocked_bindings(...) at tests/testthat/helper-TealAppDriver.R:7:3
  4. |   \-TealAppDriver$new(...) at tests/testthat/helper-TealAppDriver.R:10:7
  5. |     \-teal (local) initialize(...)
  6. |       +-base::suppressWarnings(...)
  7. |       | \-base::withCallingHandlers(...)
  8. |       \-super$initialize(...)
  9. |         \-shinytest2:::app_initialize(...)
 10. |           +-base::withCallingHandlers(...)
 11. |           \-shinytest2:::app_initialize_(self, private, ..., view = view)
 12. |             +-base::withCallingHandlers(...)
 13. |             \-self$wait_for_idle(duration = 200, timeout = private$load_timeout)
 14. |               \-shinytest2:::app_wait_for_idle(...)
 15. |                 \-shinytest2:::app_abort(self, private, "An error occurred while waiting for Shiny to be stable")
 16. |                   \-rlang::abort(..., app = self, call = call)
 17. |                     \-rlang:::signal_abort(cnd, .file)
 18. |                       \-base::signalCondition(cnd)
 19. +-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 20. | \-shinytest2:::app_abort(...)
 21. |   \-rlang::abort(..., app = self, call = call)
 22. |     \-rlang:::signal_abort(cnd, .file)
 23. |       \-base::signalCondition(cnd)
 24. \-shinytest2 (local) `&lt;fn&gt;`(`&lt;rlng_rrr&gt;`)
 25.   \-shinytest2:::app_abort(...)
 26.     \-rlang::abort(..., app = self, call = call)</error>
    </testcase>
  </testsuite>
</testsuites>
